From 059a08952a48fba45b1f7d43035e39626b7c2d79 Mon Sep 17 00:00:00 2001
From: Kent Overstreet <kmo@daterainc.com>
Date: Thu, 24 Oct 2013 17:12:52 -0700
Subject: [PATCH 2/2] bcache: Fix a lockdep splat
References: bnc#893777
Git-Commit: dd9ec84da530d3a92e99ad1f52edae44533cc27f
Patch-Mainline: v3.13

bch_keybuf_del() takes a spinlock that can't be taken in interrupt context -
whoops. Fortunately, this code isn't enabled by default (you have to toggle a
sysfs thing).

Signed-off-by: Kent Overstreet <kmo@daterainc.com>
Signed-off-by: Oliver Neukum <oneukum@suse.de>
---
 drivers/md/bcache/movinggc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/bcache/movinggc.c b/drivers/md/bcache/movinggc.c
index 1a3b4f4..dd8a035 100644
--- a/drivers/md/bcache/movinggc.c
+++ b/drivers/md/bcache/movinggc.c
@@ -113,7 +113,7 @@ static void write_moving(struct closure *cl)
 		bch_insert_data(&s->op.cl);
 	}
 
-	continue_at(cl, write_moving_finish, NULL);
+	continue_at(cl, write_moving_finish, bch_gc_wq);
 }
 
 static void read_moving_submit(struct closure *cl)
-- 
1.8.4.5

